### ISSUE TITLE
Add `axis` and `where` arguments to loss functions

### ISSUE THREAD

**Issue by carlosgmartin** [google-deepmind/optax#902]
**Feature request:** Add the following arguments:
- `axis`: reduction axis (default `-1`)
- `where`: reduction mask (default `None`)

to the following [loss functions](https://optax.readthedocs.io/en/latest/api/losses.html):
- [ ] [`convex_kl_divergence`](https://optax.readthedocs.io/en/latest/api/losses.html#optax.convex_kl_divergence)
- [ ] [`cosine_distance`](https://optax.readthedocs.io/en/latest/api/losses.html#optax.cosine_distance)
- [ ] [`cosine_similarity`](https://optax.readthedocs.io/en/latest/api/losses.html#optax.cosine_similarity)
- [ ] [`kl_divergence`](https://optax.readthedocs.io/en/latest/api/losses.html#optax.kl_divergence)
- [ ] [`sigmoid_focal_loss`](https://optax.readthedocs.io/en/latest/api/losses.html#optax.sigmoid_focal_loss)
- [ ] [`softmax_cross_entropy`](https://optax.readthedocs.io/en/latest/api/losses.html#optax.softmax_cross_entropy)
- [ ] [`softmax_cross_entropy_with_integer_labels`](https://optax.readthedocs.io/en/latest/api/losses.html#optax.softmax_cross_entropy_with_integer_labels)

I can submit a PR for this.

**Comment by vroulet** [comment]
Hello @carlosgmartin,

Yes, that would be great. Thanks for catching this!

**Comment by carlosgmartin** [comment]
Great. Once #898 is merged I'll put together a PR.
